drivers/timer/arcv2_timer0.c

```diff
@@ -246,7 +245,8 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	u32_t delay;
 	u32_t key;
 
-	ticks = MIN(MAX_TICKS, MAX(ticks - 1, 0));
+	/* aligned to next target tick */
+	ticks = MIN(MAX_TICKS, ticks);
```

Comment:
Here also it's completely unclear why do we do what we do and why new implementation is better. And even if that change helps us to get some test running better if we don't understand real root-cause of the issue we'll hit it later in more obscure case.

Reply:
It's bug here, no need -1, because I copy old codes in no-smp case where there is operation of aligned to next tick (just like + 1) later.

Reply:
So is it a fatal bug and can we address it in a separate commit?

Reply:
changes in timer driver are alread in a single commit  "drivers: timer: fix the bug for SMP".

as it's related to smp fix,  better in this PR.